src/Sylius/Behat/Context/Ui/Admin/ManagingZonesContext.php

```diff
@@ -0,0 +1,422 @@
+     * @When /^I remove (the "([^"]*)" country member)$/
+     * @When /^I remove (the "([^"]*)" zone member)$/
+     */
+    public function iRemoveTheMember(ZoneMemberInterface $zoneMember)
+    {
+        $this->updatePage->removeMember($zoneMember);
+    }
+
+    /**
+     * @When I rename it to :name
+     */
+    public function iRenameItTo($name)
+    {
+        $this->updatePage->nameIt($name);
+    }
+
+    /**
+     * @When I name it :name
+     */
+    public function iNameIt($name)
+    {
+        $this->createPage->nameIt($name);
+    }
+
+    /**
+     * @When I specify its code as :code
+     */
+    public function iSpecifyItsCodeAs($code)
+    {
+        $this->createPage->specifyCode($code);
+    }
+
+    /**
+     * @When I do not specify its code
+     */
+    public function iDoNotSpecifyItsCode()
+    {
+        // Intentionally left blank to fulfill context expectation
+    }
+
+    /**
+     * @When I do not specify its name
+     */
+    public function iDoNotSpecifyItsName()
+    {
+        // Intentionally left blank to fulfill context expectation
+    }
+
+    /**
+     * @When I do not add a country member
+     */
+    public function iDoNotAddACountryMember()
+    {
+        // Intentionally left blank to fulfill context expectation
+    }
+
+    /**
+     * @When I add a country :name
+     * @When I add a province :name
+     * @When I add a zone :name
+     */
+    public function iAddACountry($name)
+    {
+        $this->createPage->addMember();
+        $this->createPage->chooseMember($name);
+    }
+
+    /**
+     * @When I add it
+     * @When I try to add it
+     */
+    public function iAddIt()
+    {
+        $this->createPage->create();
+    }
+
+    /**
+     * @When I save my changes
+     */
+    public function iSaveMyChanges()
+    {
+        $this->updatePage->saveChanges();
+    }
+
+    /**
+     * @Then I should be notified that it has been successfully created
+     */
+    public function iShouldBeNotifiedAboutSuccessfulCreation()
+    {
+        $this->notificationChecker->checkCreationNotification(self::RESOURCE_NAME);
+    }
+
+    /**
+     * @Then I should be notified that it has been successfully edited
+     */
+    public function iShouldBeNotifiedAboutSuccessfulEdition()
+    {
+        $this->notificationChecker->checkEditionNotification(self::RESOURCE_NAME);
+    }
+
+    /**
+     * @Then I should be notified that it has been successfully deleted
+     */
+    public function iShouldBeNotifiedThatItHasBeenSuccessfullyDeleted()
+    {
+        $this->notificationChecker->checkDeletionNotification(self::RESOURCE_NAME);
+    }
+
+    /**
+     * @Then /^the (zone named "[^"]+") with (the "[^"]+" country member) should appear in the registry$/
+     * @Then /^the (zone named "[^"]+") with (the "[^"]+" province member) should appear in the registry$/
+     * @Then /^the (zone named "[^"]+") with (the "[^"]+" zone member) should appear in the registry$/
```

Comment:
`@Then /^the (zone named "[^"]+") with (the "[^"]+" (?:country|province|zone) member) should appear in the registry$/`

wdyt? 🎋